### PR TITLE
Handle missing natural inflow data

### PR DESCRIPTION
**Pull request by llavin13**
meant to do this a couple weeks back but now have a full set of cases to translate...

Purpose here is to have handling for if natural_inflow property is missing to fill it with 0's instead of throwing an error. Just opening PR for now - will clean up soon.

### CODE REVIEW

**Review by GordStephen — CHANGES_REQUESTED** [review]
I try to avoid using exception handling for control flow when simpler conditional statements would work. In this case, you can use `haskey` to check whether the dataset exists (this is actually exactly what the index lookup does internally to check whether it needs to throw an exception in the first place: https://github.com/JuliaIO/HDF5.jl/blob/5d826687c016cd56397c66029ec24a0087463650/src/highlevel.jl#L75).

**Comment by GordStephen** [comment]
Also, since your second and third commits just cancel each other out, you should probably just eliminate them. A few options for doing that:

1 - You could switch back to master then run `git branch --force ll/natural-inflow 6c522a4` - this just changes the commit that the `ll/natural-inflow` branch "points" to
2 - Staying on the `ll/natural-inflow` branch, you could `git reset HEAD~2` to step back two commits (and move the branch pointer with you)
3 - Since I'm requesting changes anyways, make those changes then run `git rebase -i master`, which lets you rewrite the history of everything that's happened since you diverged from `master`. You can then drop the second and third commits, and squash the first and (not yet existing) fourth together into a single commit

**Comment by llavin13** [comment]
uh, so there are changes but I am ultimately confused. It thought I did the rebase, but, I can't figure out how to push it to this branch without conflicts?

**Comment by GordStephen** [comment]
Oh, yeah, sorry I should have flagged that - conflicts are normal when you rebase since you're essentially rewriting history, so you'll have to force push. As long as it's just on your feature branch there's nothing wrong with it (and if you want to be super-safe you can always check out another copy of the branch before doing that so you have a "backup" - although you don't need it, since the old commits will still be around just without a human-readable name associated with them).

File: src/process_generators_storages.jl [hunk 1]
```diff
@@ -77,8 +77,11 @@ function process_generators_storages!(
 
         end
 
-        raw_inflows =
-            readsingleband(plexosfile["/data/ST/interval/storages/Natural Inflow"])
+        if haskey(plexosfile, "/data/ST/interval/storages/Natural Inflow") 
+            raw_inflows = readsingleband(plexosfile["/data/ST/interval/storages/Natural Inflow"])
+        else
+            raw_inflows = zeros(size(raw_chargeefficiency)[1],size(raw_chargeefficiency)[2])
```
Comment (GordStephen) [inline src/process_generators_storages.jl:83]: This can actually just be `zeros(size(raw_chargeefficiency))`

**Review by GordStephen — APPROVED** [review]

**Pull request merged**